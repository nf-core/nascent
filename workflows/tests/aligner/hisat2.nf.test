nextflow_pipeline {

    name "HISAT2"
    script "../../../main.nf"

    test("Should run with defaults") {

        tag "HISAT2"

        when {
            params {
                outdir = "$outputDir"
                aligner = "hisat2"
                hisat2_index = 'https://raw.githubusercontent.com/nf-core/test-datasets/nascent/reference/GRCh38_chr21_hisat2.tar.gz'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml")).match("software_versions") },
                { assert snapshot(
                    workflow.trace.tasks().size(),
                    path("$outputDir/bbmap").list(),
                    // BUG https://github.com/nf-core/nascent/issues/102
                    // path("$outputDir/bedtools").list(),
                    // FIXME gtf2bed isn't deterministic
                    // path("$outputDir/bwa/featurecounts/gene").list(),
                    // FIXME cd4 isn't deterministic
                    // path("$outputDir/bwa/featurecounts/predicted").list(),
                    // path("$outputDir/deeptools").list(),
                    // path("$outputDir/fastp/*.json").list(),
                    // FIXME gtf2bed isn't deterministic
                    // path("$outputDir/gtf2bed").list(),
                    path("$outputDir/homer/cd4.bed"),
                    path("$outputDir/homer/jurkat.bed"),
                    // FIXME https://github.com/nf-core/nascent/issues/148
                    // path("$outputDir/preseq").list(),
                    // path("$outputDir/rseqc").list(),
                    bam("$outputDir/samtools/cd4_REP1.sorted.bam").getSamLinesMD5(),
                    bam("$outputDir/samtools/cd4_REP2.sorted.bam").getSamLinesMD5(),
                    bam("$outputDir/samtools/jurkat.sorted.bam").getSamLinesMD5(),
                ).match("output_files")
                }
            )
        }
    }
}
