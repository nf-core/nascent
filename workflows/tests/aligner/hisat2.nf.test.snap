{
    "output_files": {
        "content": [
            77,
            [
                "cd4_REP1.coverage.hist.txt:md5,dff9783a6054f490bd106a5ea59ff323",
                "cd4_REP1.coverage.stats.txt:md5,bd2ae14e43e59332d62af60601f95368",
                "cd4_REP2.coverage.hist.txt:md5,ca9e862c16d556b3f1baa230d565e0a7",
                "cd4_REP2.coverage.stats.txt:md5,14d8e9d53e129de3caf98fe6c8f52ace",
                "jurkat.coverage.hist.txt:md5,25ff1313ce15f2332b3b9ac5e440ad76",
                "jurkat.coverage.stats.txt:md5,e27a85bf418ceec8205c3ed4c9e9c0e4"
            ],
            "cd4.bed:md5,d8c78818eb1666575ef4c2534ea3727e",
            "jurkat.bed:md5,7298daa579135dfb8924067abaa2ba4e",
            "e4a8dd8442a85921d5b9a7f2fd1fce1e",
            "4dba3f03e302b5d4a31c50987844ced7",
            "2f7a17a247d070580890c62005d261fa"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-03T10:31:53.720198479"
    },
    "software_versions": {
        "content": [
            "{BBMAP_PILEUP={bbmap=39.01, samtools=1.16.1, pigz=2.6}, BEDTOOLS_GENOMECOV_MINUS={bedtools=2.31.1}, BEDTOOLS_GENOMECOV_PLUS={bedtools=2.31.1}, BEDTOOLS_INTERSECT={bedtools=2.31.1}, BEDTOOLS_INTERSECT_FILTER={bedtools=2.31.1}, DEEPTOOLS_BAMCOVERAGE_MINUS={deeptools=3.5.1}, DEEPTOOLS_BAMCOVERAGE_PLUS={deeptools=3.5.1}, FASTP={fastp=0.23.4}, FASTQC={fastqc=0.12.1}, GTF2BED={perl=5.26.2}, HISAT2_ALIGN={hisat2=2.2.1, samtools=1.16.1}, HOMER_MAKETAGDIRECTORY={homer=4.11, samtools=1.11}, PINTS_CALLER={python=3.10.6, pints=1.1.8}, PRESEQ_CCURVE={preseq=3.1.1}, PRESEQ_LCEXTRAP={preseq=3.1.1}, RSEQC_INFEREXPERIMENT={rseqc=5.0.2}, RSEQC_READDISTRIBUTION={rseqc=5.0.2}, RSEQC_READDUPLICATION={rseqc=5.0.2}, SAMTOOLS_FLAGSTAT={samtools=1.2}, SAMTOOLS_IDXSTATS={samtools=1.2}, SAMTOOLS_INDEX={samtools=1.2}, SAMTOOLS_SORT={samtools=1.2}, SAMTOOLS_STATS={samtools=1.2}, SUBREAD_FEATURECOUNTS_GENE={subread=2.0.1}, UNTAR_HISAT2_INDEX={untar=1.34}, Workflow={nf-core/nascent=v2.3.0dev}}"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-04T20:20:46.356464865"
    }
}